core/vm/ipgraph.go

```diff
@@ -44,25 +81,25 @@ func (c *ipGraph) Run(evm *EVM, input []byte) ([]byte, error) {
 
 	switch {
 	case bytes.Equal(selector, addParentIpSelector):
-		return c.addParentIp(args, evm)
+		return c.addParentIp(args, evm, ipGraphAddress)
 	case bytes.Equal(selector, hasParentIpSelector):
-		return c.hasParentIp(args, evm)
+		return c.hasParentIp(args, evm, ipGraphAddress)
 	case bytes.Equal(selector, getParentIpsSelector):
-		return c.getParentIps(args, evm)
+		return c.getParentIps(args, evm, ipGraphAddress)
 	case bytes.Equal(selector, getParentIpsCountSelector):
-		return c.getParentIpsCount(args, evm)
+		return c.getParentIpsCount(args, evm, ipGraphAddress)
 	case bytes.Equal(selector, getAncestorIpsSelector):
-		return c.getAncestorIps(args, evm)
+		return c.getAncestorIps(args, evm, ipGraphAddress)
 	case bytes.Equal(selector, getAncestorIpsCountSelector):
-		return c.getAncestorIpsCount(args, evm)
+		return c.getAncestorIpsCount(args, evm, ipGraphAddress)
 	case bytes.Equal(selector, hasAncestorIpsSelector):
-		return c.hasAncestorIp(args, evm)
+		return c.hasAncestorIp(args, evm, ipGraphAddress)
 	case bytes.Equal(selector, setRoyaltySelector):
-		return c.setRoyalty(args, evm)
+		return c.setRoyalty(args, evm, ipGraphAddress)
 	case bytes.Equal(selector, getRoyaltySelector):
-		return c.getRoyalty(args, evm)
+		return c.getRoyalty(args, evm, ipGraphAddress)
 	case bytes.Equal(selector, getRoyaltyStackSelector):
-		return c.getRoyaltyStack(args, evm)
+		return c.getRoyaltyStack(args, evm, ipGraphAddress)
```

Comment:
Do they change the states of ipgraph.go?

Reply:
@kingster-will Are these updates changing the states?

Reply:
nope, it doesn't change the existing iliad states.

Reply:
@kingster-will Should we not add ipGraphAddress in the existing function parameters if they do not change states?

Reply:
Adding parameter "ipGraphAddress"  is for reuse function to avoid duplication. and these are precompile internal functions.